src/__integration__/Integration.test.ts

```diff
@@ -0,0 +1,221 @@
+import { PPTPortTypes } from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+
+import { left } from "fp-ts/lib/Either";
+import { IResponseSuccessJson } from "italia-ts-commons/lib/responses";
+import "jest-xml-matcher";
+import { PagoPaConfig } from "../Configuration";
+import {
+  activatePaymentToPagoPa,
+  checkPaymentToPagoPa
+} from "../controllers/restful/PaymentController";
+import {
+  CodiceContestoPagamento,
+  FiscalCode,
+  Importo,
+  IUV
+} from "../types/CommonTypes";
+import { PaymentsActivationRequest } from "../types/controllers/PaymentsActivationRequest";
+import { PaymentsActivationResponse } from "../types/controllers/PaymentsActivationResponse";
+import { PaymentsCheckRequest } from "../types/controllers/PaymentsCheckRequest";
+import { PaymentsCheckResponse } from "../types/controllers/PaymentsCheckResponse";
+import {
+  createPagamentiTelematiciPspNodoClient,
+  FakePagamentiTelematiciPspNodoAsyncClient
+} from "./fake/fakePagamentiTelematiciPspNodoAsyncClient";
+import mockReq from "./fake/request";
+
+const aConfig = {
+  HOST: process.env.PAGOPA_HOST || "http://localhost",
+  PORT: process.env.PAGOPA_PORT || "3001",
+  SERVICES: {
+    PAYMENTS_CHECK: "nodoVerificaRPT",
+    PAYMENTS_ACTIVATION: "nodoAttivaRPT"
+  },
+  IDENTIFIER: {
+    IDENTIFICATIVO_PSP: "AGID_01",
+    IDENTIFICATIVO_INTERMEDIARIO_PSP: "97735020584",
+    IDENTIFICATIVO_CANALE: "97735020584_02",
+    TOKEN: process.env.PAGOPA_TOKEN || "ND"
+  }
+};
+
+const aCodiceContestoPagamento = "05245c90-7468-11e8-b9bf-91897339427e" as CodiceContestoPagamento;
+
+describe("checkPaymentToPagoPa", async () => {
+  it("should return the right response", async () => {
+    const aPaymentCheckRequest: PaymentsCheckRequest = {
+      codiceIdRPT: {
+        CF: "DVCMCD99D30E611V" as FiscalCode,
+        AuxDigit: "0",
+        CodIUV: "010101010101010" as IUV,
+        CodStazPA: "22"
+      }
+    };
+
+    const verificaRPTPagoPaClient = new FakePagamentiTelematiciPspNodoAsyncClient(
+      await createPagamentiTelematiciPspNodoClient({
+        envelopeKey: PPTPortTypes.envelopeKey
+      })
+    );
+
+    const req = mockReq();
+
+    req.params = aPaymentCheckRequest;
+
+    const aPaymentCheckResponse = (await checkPaymentToPagoPa(
+      aConfig as PagoPaConfig,
+      verificaRPTPagoPaClient
+    )(req)) as IResponseSuccessJson<PaymentsCheckResponse>;
```

Comment:
you don't have to cast here but check that `isRight(errorOrPaymentCheckResponse )`
then `if (isRight(errorOrPaymentCheckResponse)) { expect(errorOrPaymentCheckResponse.value) ... }`